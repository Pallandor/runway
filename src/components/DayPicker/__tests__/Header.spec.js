import React from 'react';
import { shallow } from 'enzyme';
import Header from '../components/Header';

describe('Header', () => {
  let component;

  it('renders correctly with defaults', () => {
    component = shallow(
      <Header
        closeDialog={() => {}}
        rowStyles={{}}
        dayLabels={['Mon', 'Tue']}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with props provided', () => {
    component = shallow(
      <Header
        headerLabel="header label"
        closeAriaLabel="close"
        closeDialog={() => {}}
        rowStyles={{}}
        dayLabels={['Mon', 'Tue', 'Wed']}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
