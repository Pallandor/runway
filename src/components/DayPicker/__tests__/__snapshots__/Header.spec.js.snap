// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly with defaults 1`] = `
.emotion-7 {
  z-index: 1;
}

.emotion-3 {
  font-family: "Ciutadella-Md",sans-serif;
  background: #323232;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 0 0 12px;
}

.emotion-0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  justify-self: left;
}

.emotion-2 {
  background: none;
  height: 48px;
  padding: 0;
  line-height: 0;
  border: none;
  width: 52px;
  cursor: pointer;
}

.emotion-1 {
  fill: white;
}

.emotion-6 {
  width: 100%;
  padding: 8px 0;
  font-family: "Ciutadella",sans-serif;
  font-size: 0.875rem;
  line-height: 0.875rem;
  text-align: center;
  background: #fafafa;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.08),0 0 1px 0 rgba(0,0,0,0.04);
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.emotion-4::after {
  content: "";
  position: absolute;
  right: -1px;
  height: 15px;
  border-left: 1px solid #eeeeee;
}

<div
  className="emotion-7"
>
  <div
    className="emotion-3"
  >
    <span
      className="emotion-0"
    />
    <button
      aria-label=""
      className="emotion-2"
      onClick={[Function]}
      type="button"
    >
      <CrossIcon
        className="emotion-1"
        height="32px"
        width="32px"
      />
    </button>
  </div>
  <div
    className="emotion-6"
  >
    <div
      className="emotion-4"
      key="Mon"
    >
      Mon
    </div>
    <div
      className="emotion-4"
      key="Tue"
    >
      Tue
    </div>
  </div>
</div>
`;

exports[`Header renders correctly with props provided 1`] = `
.emotion-8 {
  z-index: 1;
}

.emotion-3 {
  font-family: "Ciutadella-Md",sans-serif;
  background: #323232;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 0 0 12px;
}

.emotion-0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  justify-self: left;
}

.emotion-2 {
  background: none;
  height: 48px;
  padding: 0;
  line-height: 0;
  border: none;
  width: 52px;
  cursor: pointer;
}

.emotion-1 {
  fill: white;
}

.emotion-7 {
  width: 100%;
  padding: 8px 0;
  font-family: "Ciutadella",sans-serif;
  font-size: 0.875rem;
  line-height: 0.875rem;
  text-align: center;
  background: #fafafa;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.08),0 0 1px 0 rgba(0,0,0,0.04);
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.emotion-4::after {
  content: "";
  position: absolute;
  right: -1px;
  height: 15px;
  border-left: 1px solid #eeeeee;
}

<div
  className="emotion-8"
>
  <div
    className="emotion-3"
  >
    <span
      className="emotion-0"
    >
      header label
    </span>
    <button
      aria-label="close"
      className="emotion-2"
      onClick={[Function]}
      type="button"
    >
      <CrossIcon
        className="emotion-1"
        height="32px"
        width="32px"
      />
    </button>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-4"
      key="Mon"
    >
      Mon
    </div>
    <div
      className="emotion-4"
      key="Tue"
    >
      Tue
    </div>
    <div
      className="emotion-4"
      key="Wed"
    >
      Wed
    </div>
  </div>
</div>
`;
